geoindex: use circumference of earth as default parameters

otan opened the pull request:
Use bounds for the index as a number similar to the circumference of the
Earth. This is roughly in line for the same value used for SRID 3857,
which represents the entire planet.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54364)
<!-- Reviewable:end -->

otan committed:
geoindex: use circumference of earth as default parameters

Use bounds for the index as a number similar to the circumference of the
Earth. This is roughly in line for the same value used for SRID 3857,
which represents the entire planet.

Release note: None

otan commented:
cc @sumeerbhola what do you think of this?
i think it matches the way we use indexing with the PK & SRID 3857 well, and I think this should be in before next release. anyone who's added an index already won't be affected since it's stored on the descriptor.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54364#-:-MHSKVivEw_9zHy2jocz:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
tftr!

bors r=sumeerbhola
